components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```
@@ -243,13 +244,19 @@ public static String getLoginPageURL(String clientId, String sessionDataKey,
             String selfPath = "/oauth2/authorize";
             AuthenticationRequest authenticationRequest = new AuthenticationRequest();
 
+            int tenantId = OAuth2Util.getClientTenatId();
+
             //Build the authentication request context.
             authenticationRequest.setCommonAuthCallerPath(selfPath);
             authenticationRequest.setForceAuth(forceAuthenticate);
             authenticationRequest.setPassiveAuth(checkAuthentication);
             authenticationRequest.setRelyingParty(clientId);
-            authenticationRequest.addRequestQueryParam(FrameworkConstants.RequestParams.TENANT_ID,
-                    new String[]{String.valueOf(OAuth2Util.getClientTenatId())});
+            try {
+                authenticationRequest.setTenantDomain(OAuth2Util.getTenantDomain(tenantId));
+            } catch (IdentityOAuth2Exception e) {
+                log.error("Error while getting tenant domain from tenant id", e);
```

Maybe we can just throw the exception and be done with it rather than using (log & throw) both.